appcelerator-archive/amp
# [Monitoring] ampbeat nats implementation

## How to test
    $ ampmake build
    $ amp cluster create
    $ docker tag appcelerator/ampbeat:local localhost:5000/appcelerator/ampbeat:local
    $ docker push localhost:5000/appcelerator/ampbeat:local
    $ docker tag appcelerator/agent:local localhost:5000/appcelerator/agent:local
    $ docker push localhost:5000/appcelerator/agent:local
    $ alias mgr='docker run -it --rm --network=hostnet -v $PWD/stacks:/stacks docker --host=m1'
    $ mgr stack deploy -c /stacks/monitoring.stack.yml monitoring

Open a browser and follow these steps:
 * Go to http://localhost:5601, you should get to the management tab.
 * Replace the index name from `logstash-*` to `ampbeat-*`
 * Click outside the field to refresh
 * Select `@timestamp` in the `Time-field name` drop down list
 * Click on the create button
 * Now, click on the discover tab and you should be able to see the logs entries

The Kibana image will be updated later to include a default configuration:
 - use the `ampbeat-*` index by default
 - provide default dashboards

added kibana dashboad (json) in cmd/ampbeat to be include in amp kibana image
To use it in the meantime, import it in kibana using menu "Management" -> "Saved Objects" and then import button and load file cmd/ampbeat/kibana-amp-dashboad.json
it shows stats cpu, mem, io, net by node, service, stack and global
